simulariumio/file_converter.py

```diff
@@ -19,24 +21,207 @@
 class FileConverter(TrajectoryConverter):
     def __init__(self, input_file: InputFileData):
         """
-        This object loads the data in .simularium JSON format
-        at the input file
+        This object loads data from the input file in .simularium format.
 
         Parameters
         ----------
         input_file: InputFileData
-            A InputFileData object containing a string path
-            or string contents for the .simularium JSON file to load
+            A InputFileData object containing .simularium data to load
         """
-        print("Reading Simularium JSON -------------")
-        buffer_data = json.loads(input_file.get_contents())
+        try:
+            print("Reading Simularium JSON -------------")
+            buffer_data = json.loads(input_file.get_contents())
+        except UnicodeDecodeError:
+            print("Reading Simularium binary -------------")
+            buffer_data = FileConverter._load_binary(input_file)
         if (
             int(buffer_data["trajectoryInfo"]["version"])
             < CURRENT_VERSION.TRAJECTORY_INFO
         ):
-            buffer_data = self.update_trajectory_info_version(buffer_data)
+            buffer_data = FileConverter.update_trajectory_info_version(buffer_data)
         self._data = TrajectoryData.from_buffer_data(buffer_data)
 
+    @staticmethod
+    def _read_binary_file(input_file: InputFileData) -> Dict[str, Any]:
+        """
+        Read a .simularium binary file and return multiple views of the data
+        """
+        result = {}
+        with open(input_file.file_path, "rb") as open_binary_file:
+            result["byte_view"] = open_binary_file.read()
+            result["n_values"] = int(
+                len(result["byte_view"]) / BINARY_SETTINGS.BYTES_PER_VALUE
+            )
+            result["int_view"] = struct.unpack(
+                f'<{result["n_values"]}i', result["byte_view"]
+            )
+            result["float_view"] = struct.unpack(
+                f'<{result["n_values"]}f', result["byte_view"]
+            )
+        return result
+
+    @staticmethod
+    def _binary_header(binary_data: Dict[str, Any]) -> Dict[str, Any]:
+        """
+        Parse header of data from a .simularium binary file
+        """
+        pos = len(BINARY_SETTINGS.HEADER)
```

Comment:
a tiny bit more naming confusion: this function is called _binary_header and we are parsing this thing called BINARY_SETTINGS.HEADER and comparing it with a variable called binary_id.  I suggest the BINARY_SETTINGS.HEADER be renamed to BINARY_SETTINGS.FILE_IDENTIFIER or something like that.